src/hooks/postsCustomHooks.tsx

```diff
@@ -40,10 +41,18 @@ export const useFetchPosts = (refresh: boolean = false) => {
     const fetchPosts = async () => {
       try {
         setLoading(true);
+        const user = await getCurrentUser();
         const communityId = getCurrentCommunityID();
         const response = await getCommunityPostsAPI(communityId);
         const jsonPosts = JSON.parse(JSON.stringify(response));
-        setPosts(jsonPosts.data.getCommunity.posts.items);
+        const visiblePosts = jsonPosts.data.getCommunity.posts.items.filter((post: Post) => {
+          return (
+            post.visibility === 'PUBLIC' ||
+            post.author.id == user!.id ||
+            (post.visibility === 'PRIVATE' && user!.friends!.includes(post.author.id))
```

Comment:
I think we have FriendsOnly enum for this instead of Private, change to that

Reply:
oh yeah good catch